src/SingleSignatureDirtyHack.sol

```diff
@@ -120,8 +120,8 @@ abstract contract SingleSignatureDirtyHack is IERC5267, AbstractContext {
     // allowance.
     function _consumeNonce(address owner, uint256 incomingNonce) private {
         uint256 currentNonce = nonces[owner];
-        nonces[owner] = incomingNonce;
-        if (incomingNonce <= currentNonce) {
+        nonces[owner] = incomingNonce + 1;
```

Comment:
can this be unchecked? do we need different range bounds on the nonce to avoid overflowing the valid nonce range, smaller than the uint256 range?

Reply:
I thought that was intentional, I would say the standard is the nonces mapping just to admit any nonce in any order. Fine with that?

Reply:
> that was intentional
that it's possible for the user to grief themself? I wouldn't say intentional, but I also don't see a good solution.

do we need to restrict the range of acceptable nonces (i.e., <= type(uint64).max - 1)? can we use `unchecked` here?

Reply:
On one side, we can safely decrease to uint64, but we don't need to. We can do unchecked and revert if it becomes 0 (overflow) in the current implementation.

As for the griefing I see 2 options:
1. You assume it is incremental and every transaction is going to be consumed right after sending the approve one so signing nonce X invalidates all previous ones (current implementations)
2. You assume the `transferFrom` execution in our contract can be done unordered, then allow unordered nonces.

Number 2 can be the user submitting multiple approve trick transactions, and then "consuming" them with transferFrom in any order.

Guess this will also be improved (or removed) once we check receipts/inclusion.

Reply:
Actually it doesn't makes sense to consume unordered